File: tests/list/list.cpp
#include <bits/stdc++.h>
#include "list/list.hpp"
using namespace std;

int main() {
  pkr::List<int> l;
  size_t n;
  cin >> n;
  for (size_t i = 0; i < n; ++i) {
    char cmd;
    cin >> cmd;
    if (cmd == 's') {
      cout << l.size() << endl;
    } else if (cmd == 'c') {
      l.clear();
    } else if (cmd == 'f') {
      int k;
      cin >> k;
      l.push_front(k);
    } else if (cmd == 'p') {
      const auto vec = l.to_vector();
      copy(vec.begin(), vec.end(),
        ostream_iterator<decltype(vec)::value_type>(cout, " "));
      cout << endl;
    } else if (cmd == 'b') {
      int k;
      cin >> k;
      l.push_back(k);
    } else if (cmd == 'r') {
      int k;
      cin >> k;
      l.remove(k);
    }
  }
}

File: tests/list/list.py
n = int(input())
l = []
for i in range(n):
  cmd = input()
  if cmd[0] == 'c':
    l = []
  elif cmd[0] == 's':
    print(len(l))
  elif cmd[0] == 'p':
    for x in l:
      print(x, end = ' ')
    print()
  else:
    x = int(cmd[2:])
    if cmd[0] == 'f':
      l = [x] + l
    elif cmd[0] == 'b':
      l = l + [x]
    elif cmd[0] == 'r':
      if x in l:
        l.remove(x)
